doc/README.tpl.md

```diff
@@ -598,6 +600,51 @@ defineSupportCode(({ When }) => {
 })
 ```
 
+## Helpers
+
+### Cast helper
+
+Cast helper can be used to cast values for custom gherkin rules.
+To find more about casting see [Type System](#type-system).
+
+#### Cast usage
+
+This must be used on gherkin arrays. Based on your array type you have to use: 
+ * ```step.hashed()``` -> ```Cast.objects(step.hashes())```
+ * ```step.rows()``` -> ```Cast.array(step.rows())```
+ * ```step.raw()``` -> ```Cast.array(step.raw())```
+ * ```step.rowsHash()``` -> ```Cast.objects(step.rowsHash())```
+ 
+For example:
+```javascript
+    const { cast } = require('@ekino/veggies')
+
+    Then(/^User data should be$/, (step) => {
+        const userData = this.userData
+        const expectedData = Cast.objects(step.rowsHash())
+        expect(userData).to.be.deep.equal(expectedData)
+    })
+```
+
+#### Add a type
+
+You can provide your own type.
+For example: 
+```javascript
+    Cast.addType('newType', value => value === 'true')
```

Comment:
Please can your remove extra indentation to be homogenous with the rest of the doc?